lib/theme_check/tags.rb

```diff
@@ -169,14 +169,32 @@ class Javascript < Liquid::Raw; end
 
     class Stylesheet < Liquid::Raw; end
 
-    Liquid::Template.register_tag('form', Form)
-    Liquid::Template.register_tag('layout', Layout)
-    Liquid::Template.register_tag('render', Render)
-    Liquid::Template.register_tag('paginate', Paginate)
-    Liquid::Template.register_tag('section', Section)
-    Liquid::Template.register_tag('style', Style)
-    Liquid::Template.register_tag('schema', Schema)
-    Liquid::Template.register_tag('javascript', Javascript)
-    Liquid::Template.register_tag('stylesheet', Stylesheet)
+    class << self
+      attr_writer :register_tags
+
+      def register_tags?
+        @register_tags
+      end
+
+      def register_tag(name, klass)
+        return if Liquid::Template.tags[name]
```

Comment:
I added this line in the private repo to help Feken, but I see now that it breaks a few things when scanning Dawn. Can you remove it? Or I can push a commit with details if you prefer.

Reply:
@veken1199 what do we need here for our promote draft checks?

Reply:
Regardless, we should be fine removing this line of code since Core is depending on this implementation: 
```
return if !register_tags? || (defined?(@registered_tags) && @registered_tags)
```

Reply:
This is what I am currently doing in Core
https://github.com/Shopify/shopify/blob/819186d4f6cecdbdeee2ddef2742d0b2ee9367f1/components/online_store/app/models/online_store/app_extension/theme_check_runner.rb#L7